src/compute-client/src/protocol/command.rs

```diff
@@ -269,6 +269,25 @@ pub enum ComputeCommand<T = mz_repr::Timestamp> {
     },
 }
 
+impl ComputeCommand {
+    /// Returns a string representation of the command kind.
+    pub fn kind_as_str(&self) -> &'static str {
+        use ComputeCommand::*;
+        match self {
+            CreateTimely { .. } => "CreateTimely",
+            CreateInstance(..) => "CreateInstance",
+            InitializationComplete { .. } => "InitializationComplete",
+            AllowWrites { .. } => "AllowWrites",
+            UpdateConfiguration(..) => "UpdateConfiguration",
+            CreateDataflow(..) => "CreateDataflow",
+            Schedule(..) => "Schedule",
+            AllowCompaction { .. } => "AllowCompaction",
+            Peek(..) => "Peek",
+            CancelPeek { .. } => "CancelPeek",
+        }
+    }
+}
```

Comment:
I wonder if we need this. We already have precedent for metrics labeled by command type, for example `mz_compute_controller_history_command_count`. They use the `CommandMetrics` helper to maintain a metric per compute command type and we may be able to use that here too. The benefit would be that the metric values for all command types would be initialized to zero, vs. only initialized when the first command of that kind is observed.

If `CommandMetrics` can't easily be made to work here that's fine, but could you then change the strings here to snake-case so they are consistent with the ones produced by existing metrics? Maybe also rename this method to `as_metric_label` or something like that.

Reply:
Oh nice, I wasn't aware! I'll update the PR to use the existing infrastructure.